libvirt/resource_libvirt_volume.go

```diff
@@ -143,12 +143,11 @@ func resourceLibvirtVolumeCreate(d *schema.ResourceData, meta interface{}) error
 
 		// update the image in the description, even if the file has not changed
 		size, err := img.Size()
-		if err != nil {
-			return err
+		if err == nil {
```

Comment:
if we do so, we are not checking and returning the error of the above function 
or i am missing something here?  :thinking:

Reply:
At the moment there's not many other failure scenarios but I see the general point. If this is the only thing holding back merging just +1 this and I will fix it of course.

Reply:
@moio thanks, yop otherwise looks clean to me :guitar:

Reply:
I like how you've pushed the size information down into the copier (most of a2e28a3).  But I'd rather leave this particular hunk as it stands.  To have `httpImage.Size()` support endpoints that 403 `HEAD`:

```console
$ curl -sLI https://github.com/moio/sumaform-images/releases/download/4.0.0/centos7.qcow2 | grep 'HTTP\|Location'
HTTP/1.1 302 Found
Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/...
HTTP/1.1 403 Forbidden
```

how about a "you can't make me read the body" `GET` ;) :

```go
func (i *httpImage) Size() (uint64, error) {
  response, err := http.Head(i.url.String())
  if err != nil {
    return 0, err
  }
  if response.StatusCode == 403 {
    response, err = http.Get(i.url.String())
    if err != nil {
      return 0, err
    }
    response.Body.Close()
  }
  if response.StatusCode != 200 {
    ...
  }
  ...
}
```

Reply:
nice thx @wking  and @moio for this :japan: